sign/bdn/bdn.go

```diff
@@ -61,14 +61,39 @@ func hashPointToR(pubs []kyber.Point) ([]kyber.Scalar, error) {
 
 	coefs := make([]kyber.Scalar, len(pubs))
 	for i := range coefs {
-		coefs[i] = mod.NewIntBytes(out[i*16:(i+1)*16], modulus128, kyber.LittleEndian)
+		b, err := mod.NewIntBytes(out[i*16:(i+1)*16], modulus128, kyber.LittleEndian).MarshalBinary()
+		if err != nil {
+			return nil, err
+		}
+		if g.Scalar().ByteOrder() == kyber.BigEndian {
+			reverse(b, b)
+		}
+
+		coefs[i] = g.Scalar()
+		coefs[i].SetBytes(b)
```

Comment:
`g.Scalar` is of size 32 whereas the coefs in `bdn` are meant to be of size 128 bits, e.g. 16.
I think that's explaining the issues you've had so far.

Also why not just:

```suggestion
		b, err := mod.NewIntBytes(out[i*16:(i+1)*16], modulus128, g.Scalar().ByteOrder()).MarshalBinary()
		if err != nil {
			return nil, err
		}

		coefs[i] = b
```

Reply:
Although it looks cleaner, this is a breaking change, these bytes in `out` were always interpreted in little-endian mod 2^128 with no regard to the endianness of the suite.

Reply:
I think this is getting changed in #553 anyway now, no?

Reply:
Yes, I will remove that part and rebase on #553